libafl/src/inputs/value.rs

```diff
@@ -0,0 +1,271 @@
+//! Newtype pattern style wrapper for [`super::Input`]s
+
+use alloc::{string::String, vec::Vec};
+use core::{
+    fmt::Debug,
+    hash::Hash,
+    ops::{Deref, DerefMut},
+};
+
+use libafl_bolts::{generic_hash_std, rands::Rand};
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "std")]
+use {
+    libafl_bolts::{fs::write_file_atomic, Error},
+    std::{fs::File, io::Read, path::Path},
+};
+
+use super::{Input, MappedInput};
+use crate::{corpus::CorpusId, mutators::numeric::Numeric};
+
+/// Newtype pattern wrapper around an underlying structure to implement inputs
+///
+/// This does not blanket implement [`super::Input`], because for certain inputs, writing them to disk does not make sense, because they don't own their data (like [`super::MutVecInput`])
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Default)]
+pub struct ValueInput<I>(I);
+
+impl<I> From<I> for ValueInput<I> {
+    fn from(value: I) -> Self {
+        Self(value)
+    }
+}
+
+impl<I> ValueInput<I> {
+    /// Create a new [`ValueInput`]
+    pub const fn new(value: I) -> Self {
+        Self(value)
+    }
+
+    /// Extract the inner value
+    pub fn inner(self) -> I {
```

Comment:
For consistency with `std`, e.g., [`Cell::into_inner`](https://doc.rust-lang.org/std/cell/struct.Cell.html#method.into_inner), perhaps consider renaming?
```suggestion
    pub fn into_inner(self) -> I {
```